pkg/applicationserver/applicationserver.go

```diff
@@ -883,16 +883,18 @@ func (as *ApplicationServer) storeUplink(ctx context.Context, ids ttnpb.EndDevic
 		ReceivedAt: uplink.ReceivedAt,
 	}
 	for _, md := range uplink.RxMetadata {
-		cleanUplink.RxMetadata = append(cleanUplink.RxMetadata, &ttnpb.RxMetadata{
-			GatewayIdentifiers: ttnpb.GatewayIdentifiers{
-				GatewayId: md.GatewayId,
-			},
-			AntennaIndex:  md.AntennaIndex,
-			FineTimestamp: md.FineTimestamp,
-			Location:      md.Location,
-			Rssi:          md.Rssi,
-			Snr:           md.Snr,
-		})
+		if md.GetGatewayIds() != nil {
```

Comment:
Nit: turn this around: on `==`, do `continue`. Saves a nesting level.

I also wouldn't use the getter and as we access every other field directly, so it if `md` is `nil`, we're not gaining much.

Reply:
Agree with the `continue` on `== nil`, this is always the preferred style in Go.

And indeed, we also need to be more consistent about when to use getters and when to write our own nil checks and directly access fields.